game/hud/src/widgets/Crafting/components/Input/index.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * @Author: Mehuge (mehuge@sorcerer.co.uk)
+ * @Date: 2017-05-14 21:42:18
+ * @Last Modified by: Mehuge (mehuge@sorcerer.co.uk)
+ * @Last Modified time: 2017-05-17 19:17:09
+ */
+
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { client } from 'camelot-unchained';
+
+interface InputProps {
+  size?: number;
+  value?: string;
+  disabled?: boolean;
+  onChange: (value: string) => void;
+}
+
+interface InputState {
+  value: string;
+  changed: boolean;
+}
+
+class Input extends React.Component<InputProps, InputState> {
+  constructor(props: InputProps) {
+    super(props);
+    this.state = { changed: false, value: props.value };
```

Comment:
I would say it is common, not standard (there are plenty of examples in the code base), and usually depends on the numbers/length of properties being initialised.  

In the same way function arguments are placed on one line until they get too long and then are broken up onto several, I think structure initialisation should allow the same flexibility.  Ofc, initialising a large number of properties should be broken up onto several lines, but when the initialisation is short, it should be allowed to remain on one line.

Airbnb (which I know isn't the be all and end all) allows it as does our linter.

I will change it, if you insist ofc.